### PR TITLE
Add pv example to katibDB

### LINKED ISSUE #137: Katib needs to use durable storage that outlives the pod.
Katib uses two databases
- MySQL (for Katib)
- MongoDB (for ModelDB)

It doesn't look like we are using any sort of durable storage for either one.

Here's the MySQL deployment it doesn't look like its using a persistent volume
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/katib/vizier.libsonnet#L185

Here's the MongoDB deployment:
https://github.com/kubeflow/kubeflow/blob/master/kubeflow/katib/modeldb.libsonnet#L39

@YujiOshima @gaocegege Is this accurate or am I missing something?

### PR DESCRIPTION
**Pull request by YujiOshima**
For persistent data, add pv example for katib.
Related to #137 

Signed-off-by: YujiOshima <yuji.oshima0x3fd@gmail.com>

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/katib/178)

### CODE REVIEW

**Comment by jlewi** [comment]
@YujiOshima do you plan on making these changes to the Katib component as well?
https://github.com/kubeflow/kubeflow/tree/master/kubeflow/katib

File: manifests/pv/pv.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+metadata:
+  name: katib-volume
+  labels:
+    type: local
+    app: katib
+spec:
+  capacity:
+    storage: 10Gi
+  accessModes:
+    - ReadWriteMany
```
Comment (inc0) [inline manifests/pv/pv.yaml]: do we need ReadWriteMany? It's hard to maintain on non-minikube env

File: manifests/pv/pv.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  hostPath:
+    path: /data/katib
```
Comment (inc0) [inline manifests/pv/pv.yaml]: why do we need hostpath?

File: manifests/pv/pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: katib-volume
+  namespace: katib
+  labels:
+    app: katib
+spec:
+  accessModes:
```
Comment (inc0) [inline manifests/pv/pvc.yaml]: again, why not RWO?

File: manifests/pv/pvc.yaml [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - ReadWriteMany
+  resources:
+    requests:
+      storage: 10Gi
+  storageClassName: ""
```
Comment (inc0) [inline manifests/pv/pvc.yaml]: leave out storageClass, let's assume users have default

File: manifests/pv/pvc.yaml [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  selector:
+    matchLabels:
```
Comment (inc0) [inline manifests/pv/pvc.yaml]: no need for that if users have working storageclass. By the way, minikube comes with storageclass by default so it's unnecessary.

File: manifests/vizier/db/deployment.yaml [hunk 6]
```diff
@@ -28,10 +28,10 @@ spec:
...
           containerPort: 3306
-#        resources:
-#          requests:
-#            cpu: 500m
-#            memory: 500M
-#          limits:
-#            cpu: 500m
-#            memory: 500M
+        volumeMounts:
+        - name: katib-volume
```
Comment (inc0) [inline manifests/vizier/db/deployment.yaml]: I think better name would be katib-mysql or katib-db. We might need more volumes later

**Comment by YujiOshima** [comment]
@inc0 Thanks! I updated.

**Pull request merged**